packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -237,17 +237,20 @@ export async function pollTransformationStatusUntilPlanReady(jobId: string) {
     try {
         plan = await getTransformationPlan(jobId)
     } catch (error) {
+        // means API call failed
         getLogger().error(`CodeTransformation: ${CodeWhispererConstants.failedToCompleteJobNotification}`, error)
         transformByQState.setJobFailureErrorNotification(CodeWhispererConstants.failedToGetPlanNotification)
         transformByQState.setJobFailureErrorChatMessage(CodeWhispererConstants.failedToGetPlanChatMessage)
         throw new Error('Get plan failed')
     }
 
-    const planFilePath = path.join(os.tmpdir(), 'transformation-plan.md')
-    fs.writeFileSync(planFilePath, plan)
-    await vscode.commands.executeCommand('markdown.showPreview', vscode.Uri.file(planFilePath))
-    transformByQState.setPlanFilePath(planFilePath)
-    await vscode.commands.executeCommand('setContext', 'gumby.isPlanAvailable', true)
+    if (plan !== undefined) {
+        const planFilePath = path.join(transformByQState.getProjectPath(), 'transformation-plan.md')
+        fs.writeFileSync(planFilePath, plan)
+        await vscode.commands.executeCommand('markdown.showPreview', vscode.Uri.file(planFilePath))
+        transformByQState.setPlanFilePath(planFilePath)
+        await vscode.commands.executeCommand('setContext', 'gumby.isPlanAvailable', true)
+    }
     sessionPlanProgress['generatePlan'] = StepProgress.Succeeded
```

Comment:
Should this be moved into the `if` block above? If `plan` is undefined, this step didn't succeed

Reply:
No this is fine. We need to set it to succeeded so that the Transformation Hub steps continue to render properly, and technically the plan did "generate" properly, it just didn't get displayed to user due to some parsing issue. As a followup, we should technically show a yellow icon instead of a green checkmark when this happens, but that would be the only change to make.